In [ ]:
import pandas as pd
import sys
import numpy as np
from sklearn import metrics 
from sklearn.linear_model import LogisticRegression
from sklearn.model_selection import train_test_split
from sklearn import datasets
iris = datasets.load_iris()

In [ ]:

X = iris.data[:, :3] #taking first three features
Y = iris.target

X_train, X_test, Y_train, Y_test = train_test_split(X, Y, test_size=0.2, random_state=3)

1. Attempting Logistic Regression

In [ ]:
logistic_regression = LogisticRegression()

In [ ]:
logistic_regression.fit(X_train, Y_train)


LogisticRegression()

In [ ]:
Y_pred = logistic_regression.predict(X_test)

In [ ]:
accuracy = metrics.accuracy_score(Y_test, Y_pred)
accuracy_percentage = 100*accuracy
accuracy_percentage

93.33333333333333

Accuracy of the Logistic Regression model is 93.33%

2. Attempting Gaussian Naive Bayes algorithm 

In [ ]:
from sklearn.naive_bayes import GaussianNB
gnb = GaussianNB()
gnb.fit(X_train, Y_train)

GaussianNB()

In [ ]:
Y_pred = gnb.predict(X_test)

In [ ]:
from sklearn import metrics
print(metrics.accuracy_score(Y_test, Y_pred)*100)

86.66666666666667


Accuracy of the Gaussian Naive Bayes algorithm is 86.66%

**3**.Having more features does not always guarantees to improve the accuracy.  It might lead to overfitting, which doesn't respond well to new data. Increasing the number of features can also increase the learning time. If it will learn for too long, the model will become more prone to overfitting. 
Accuracy changes when more features are used to fit the models. We will have to scale the values for the iris dataset so that total number of iterations do not reach the limit.